### ISSUE TITLE
Note that Object Activity types are core rather than exhaustive

### ISSUE THREAD

**Issue by michaelwebbjisc** [jiscdev/xapi#163]
There are lots of activity types that come from VLE specific plugins, and these aren't defined in our list.  Rather than throw this data away it make sense to document that this is a list of core Object types, and that VLEs may send other platform specific object types.

These should be of the form:

http://xapi.jisc.ac.uk/define/vle

The place to document this would be in this section:

3.0 Object >3.1 Activity Types >The following are types of the Activity Definition Object.

**Comment by ds10** [comment]
Just to check:

This is this to capture data generated by plugins that we may have no idea are installed? I'm not sure how this would be captured? What do you mean that it should be in the form http://xapi.jisc.ac.uk/define/vle?

On another note to do with the uri http://xapi.jisc.ac.uk/define/vle, I had the impression we had dropped /define from the path. I've noticed a documentation bug that we use http://xapi.jisc.ac.uk/define/vle/page in example at 3.0->3.1, but not in the actual vocab in that documentation. The version changes document that kept track of uri changes (and now also template changes) does say the /define/ part of the path was  dropped in v0.2 https://github.com/jiscdev/xapi/blob/1.0/version_changes.md#vocabulary-changes.

However, in the master branch it still exists in the module-view statement template- but not in the 1.0 branch.

**Comment by michaelwebbjisc** [comment]
yes, sorry, the /define was a complete red herring!

I meant: 

http://xapi.jisc.ac.uk/vle

so in moodle for example that are activity types that we haven't talked about called 'data' and 'choice'

These are sent as:

http://xapi.jisc.ac.uk/vle/data

http://xapi.jisc.ac.uk/vle/choice

Rather than try and maintain an exhaustive list of everything that any VLE can sent as an object that can be viewed, I'm saying if it's outside our core list then so long as in the form http://xapi.jisc.ac.uk/vle/name it's fine.

**Comment by ds10** [comment]
Right ok then I understand this now. I'll update the documentation, fix the /define/ documentation bug and bring up on monday.

**Comment by arc12** [comment]
for the documentation, I suggest it is clear that statements involving off-list URIs are required to use an "experimental" version (http://xapi.jisc.ac.uk/version) in the statement

**Comment by michaelwebbjisc** [comment]
I don't think they should - the structure is identical to the on list ones.

**Comment by michaelwebbjisc** [comment]
Here's a list of the output from a typical Moodle install, just so we can see what we've got.

http://xapi.jisc.ac.uk/vle/imscp	4	Non-core
http://xapi.jisc.ac.uk/vle/forum	867	Core
http://adlnet.gov/expapi/activities/course	380789	Core
http://activitystrea.ms/schema/1.0/application	163221	Core
http://xapi.jisc.ac.uk/vle/page	6035	Core
http://xapi.jisc.ac.uk/vle/data	21574	Non-core
http://xapi.jisc.ac.uk/vle/choice	2813	Non-core
http://xapi.jisc.ac.uk/vle/module	2752	Non-core
http://xapi.jisc.ac.uk/vle/lti	30	Non-core
http://xapi.jisc.ac.uk/vle/chat	105	Non-core
http://xapi.jisc.ac.uk/vle/survey	5	Non-core
http://adlnet.gov/expapi/activities/module	10888	Core
http://xapi.jisc.ac.uk/vle/glossary	982	Non-core
http://xapi.jisc.ac.uk/vle/quiz	88421	Non-core
http://xapi.jisc.ac.uk/vle/content	127280	Non-core
http://xapi.jisc.ac.uk/externalURL	28733	Core
http://xapi.jisc.ac.uk/vle/workshop	54	Non-core
http://xapi.jisc.ac.uk/vle/wiki	8	Non-core
http://xapi.jisc.ac.uk/vle/lesson	31	Non-core
http://xapi.jisc.ac.uk/vle/scorm	2695	Non-core

We could just document these I guess, then they become core anyway.  The way the moodle plugin works at the moment is that if it comes across an unknown thing it just constructs the URI.  This means we don't have to keep getting people to upgrade the plugin to get new data (we want to avoid people have to upgrade their plugin as much as possible).  Given that these are all part of 'viewing a thing on a VLE' recipe it makes no sense to me to class these as experimental.

**Comment by arc12** [comment]
I would much prefer a spec where we had a clear and cross-platform (Bb, Moodle, Canvas, ...) set of object types with the plugin being responsible for translating from local pecularities than to have the plugin fire and forget... If all the VLEs spew out mostly ideosyncratic URIs, we end up losing some of the point of creating the recipes in the first place.

Some of those appear to me to be mappable to plain-old-content. Others are likely to also have a counterpart in other VLEs, so lets include as core. A few are not at all clear and might actually be interesting, although the counts are low enough to beg the question "why bother". Two kinds of module looks unhelpful.

Clean and constrained semantics of statements emitted by the plugins is the best way to be sustainable and avoid regret, IMO.

**Comment by michaelwebbjisc** [comment]
The problem with that approach is it loses value to the institutions.  I agree what you are suggestions works best from a predictive analytics perspective, but not on straight description, where people are interested in the exact detail.  The beauty of xAPI is it allows us to accommodate this.    

The answer to 'why bother' on the low counts is that you as a lecturer might be the only one using that feature, but you'll still be really interested in how it used, so overall it's important we retain these.  That's also why we don't want to map things to plain content.

**Comment by arc12** [comment]
OK... NOW I see the requirement. Yes, I appreciate loss of fidelity is not helpful in supporting it. Still, consistency across VLEs where possible should be an aim...

**Comment by michaelwebbjisc** [comment]
Yes, agreed it would be good to make these general where we can.

This page might help: it's the list of default moodle activities:

https://docs.moodle.org/33/en/Activities

You can see in the list I posted above one completely non-standard thing:

http://xapi.jisc.ac.uk/vle/imscp  (IMS Content Package), so I'm guessing here someone was trying out a different Moodle plugin.

and here's the list for Blackboard:

https://help.blackboard.com/Learn/Instructor/Course_Content/Create_Content/Create_Course_Materials/Types_of_Course_Content

So we can see there's some crossover and a few differences.

**Comment by arc12** [comment]
... also to clarify the URI form. For example, I suggest the xapi.jisc.ac.uk namespace is kept reserved for those things which are defined in the recipes, with laissez faire object types being identified in a plugin namespace.

**Comment by ds10** [comment]
#144

**Comment by ds10** [comment]
Some domaine xpert assistance is required here (see ticket #166)

**Comment by michaelwebbjisc** [comment]
Here's my attempt:

Where it helps, I've put subtype examples.  The subtypes I've namespace to jisc, but not http://xapi.jisc.ac.uk, just because we own them rather than the vendors in this instance.

The summary is i think we retire /file and cover this by /content instead.
We add /externalcontent which covers scorm, lti etc
We add /chat, which I think is different enough from form to need it's own. 

So the subtypes aren't part of the spec, so we don't need to agree them, as we leave those to the plugin, but this is an example of what it would look like.

**1. http://activitystrea.ms/schema/1.0/application**

**2. http://adlnet.gov/expapi/activities/assessment**

Example subtypes:
http://bblearn.xapi.jisc.ac.uk/selfandpeerassessment
http://bblearn.xapi.jisc.ac.uk/safeassignment
http://moodle.xapi.jisc.ac.uk/assignment

**3. http://xapi.jisc.ac.uk/vle/forum**

**4. http://adlnet.gov/expapi/activities/course**

**http://xapi.jisc.ac.uk/vle/page**

Example subtypes:

http://moodle.xapi.jisc.ac.uk/page
http://bblearn.xapi.jisc.ac.uk/blankpage
http://bblearn.xapi.jisc.ac.uk/modulepage

**5. http://xapi.jisc.ac.uk/vle/content**

Example subtypes:

http://bblearn.xapi.jisc.ac.uk/audio
http://bblearn.xapi.jisc.ac.uk/video
http://bblearn.xapi.jisc.ac.uk/image
http://moodle.xapi.jisc.ac.uk/data
http://moodle.xapi.jisc.ac.uk/file
http://moodle.xapi.jisc.ac.uk/glossary
http://moodle.xapi.jisc.ac.uk/wiki
http://moodle.xapi.jisc.ac.uk/lesson
http://bblearn.jisc.xapi.ac.uk/lessonplan
http:/moodle.xapi.jisc.ac.uk/vle/workshop

**6. http://xapi.jisc.ac.uk/vle/chat**

**7. http://adlnet.gov/expapi/activities/module**

Example Subtypes
http://bblearn.xapi.jisc.ac.uk/learningmodule

**8. http://xapi.jisc.ac.uk/vle/quiz**

http://moodle.xapi.jisc.ac.uk/quiz
http://moodle.xapi.jisc.ac.uk/survey
http://moodle.xapi.jisc.ac.uk/choice
http://bblearn.xapi..jisc.ac.uk/test
http://bblearn.xapi.jisc.ac.uk/survey


Example subtypes:

http://moodle.xapi.jisc.ac.uk/externalURL
http://bblearn.xapi.jisc.ac.uk/webLink

**9. http://xapi.jisc.ac.uk/externalContent**

Example subtypes:

http://jisc.xapi.jisc.ac.uk/scorm
http://jisc.xapi.jisc.ac.uk/lti
http://bblearn.xapi.jisc.ac.uk/mashup

**Comment by ds10** [comment]
I've updated the vocab, adding an subtype example column to some attributes. I went to check that the examples in VLE statement templates are correct, and noticed that they are all seem wrong.. I think this is because we have gone through a few rounds of discussion, and the subtype in these examples is leftover from when we used applicationType , which I don't think ever actually made it in to a release. It is just the template.md files as the moodle and blackboard outputs seem correct.

Basically, all the examples seem to have the following in object.definition.extensions:

``` javascript
"extensions": {
      "http://xapi.jisc.ac.uk/subType": "http://id.tincanapi.com/activitytype/lms"
    }
``` 

Which doesn't make sense or fit in with what we want subtype to do? I'm guessing it is there from when we used to have an applicationType?

Just checking I have not missed anything before I remove this from the templates (or update them with a subtype that makes sense)
